src/echo-common/src/plugin-context.ts

```diff
@@ -0,0 +1,23 @@
+import { LoggingService } from "./logging-service";
+
+export type PluginMetadata = {
+  name: string
+};
+
+export type PluginContext = {
```

Comment:
https://github.com/PoeStack/poestack-sage/blob/main/src/echo-common/src/echo-context.ts#L14 this should be used instead, each plugin already has its own instance created in https://github.com/PoeStack/poestack-sage/blob/main/src/echo-app/src/renderer/src/echo-context-factory.ts which they use for these types of services. The `contextSource` should be used as the name I don't think this is set currently but should be easy enough to update I think right now it just always passes in "plugin".

Reply:
Yea I saw this too. But I thought pushing a context instead of pulling is better. That way the plugins are isolated and may not use the context of other plugins, this is important so the logging service scoping is actually working how we want :). Additionally you don't have to call the echo context service manually with a string.

But if both of you @zach-herridge & @C3ntraX think the echo context factory is a better approach, I can also go with this way. If we decide that the push option is the better one I can integrate it into the plugins.

Reply:
The issue with pushing is that it's pretty easy for things to happen in the plugin before the entry is called so if they can't access their context before the entry method they are more limited. They could access another plugins context but we wouldn't approve a PR doing that without a really good reason.

Reply:
Alright, makes sense. I changed it to use the echo context factory.